### PR TITLE
Refactor `ci_testing` and separate it from `DevToolsPlugin`

### PR DESCRIPTION
**Pull request by BD103**
# Objective

- We use [`ci_testing`](https://dev-docs.bevyengine.org/bevy/dev_tools/ci_testing/index.html) to specify per-example configuration on when to take a screenshot, when to exit, etc.
- In the future more features may be added, such as #13512. To support this growth, `ci_testing` should be easier to read and maintain.

## Solution

- Convert `ci_testing.rs` into the folder `ci_testing`, splitting the configuration and systems into `ci_testing/config.rs` and `ci_testing/systems.rs`.
- Convert `setup_app` into the plugin `CiTestingPlugin`. This new plugin is added to both `DefaultPlugins` and `MinimalPlugins`.
- Remove `DevToolsPlugin` from `MinimalPlugins`, since it was only used for CI testing.
- Clean up some code, add many comments, and add a few unit tests.

## Testing

The most important part is that this still passes all of the CI validation checks (merge queue), since that is when it will be used the most. I don't think I changed any behavior, so it should operate the same.

You can also test it locally using:

```shell
# Run the breakout example, enabling `bevy_ci_testing` and loading the configuration used in CI.
CI_TESTING_CONFIG=".github/example-run/breakout.ron" cargo r --example breakout -F bevy_ci_testing
```

---

## Changelog

- Added `CiTestingPlugin`, which is split off from `DevToolsPlugin`.
- Removed `DevToolsPlugin` from `MinimalPlugins`.

## Migration Guide

Hi maintainers! I believe `DevToolsPlugin` was added within the same release as this PR, so I don't think a migration guide is needed.

`DevToolsPlugin` is no longer included in `MinimalPlugins`, so you will need to remove it manually.

```rust
// Before
App::new()
    .add_plugins(MinimalPlugins)
    .run();

// After
App::new()
    .add_plugins(MinimalPlugins)
    .add_plugins(DevToolsPlugin)
    .run();
```

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Much better: I like this organization. This *definitely* shouldn't be part of `DevToolsPlugin`, and `DevToolsPlugin` should not be part of `MinimalPlugins`.

**Review by pablo-lua — APPROVED** [review]
Agreed here, Ci testing shouldn't be there, so all good! Though we should do something about the devtools plugin soon, but not anywhere part of this PR, LGTM

**Review by matiqo15 — APPROVED** [review]
I'm wondering whether we want to keep the `bevy_ci_testing` feature in the `bevy_dev_tools`. Overall looks good!

**Comment by BD103** [comment]
> I'm wondering whether we want to keep the `bevy_ci_testing` feature in the `bevy_dev_tools`. Overall looks good!

If we were to split it out into a separate crate, I would like it to be a bit more general purpose. I don't think its polished enough for that yet, so for now `dev_tools` is as good a place as any.

**Review by mockersf — APPROVED** [review]

File: crates/bevy_dev_tools/src/ci_testing/mod.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            ron::from_str(
+                &std::fs::read_to_string(filename)
+                    .expect("error reading CI testing configuration file"),
+            )
+            .expect("error deserializing CI testing configuration file")
+        };
+
+        #[cfg(target_arch = "wasm32")]
+        let config: CiTestingConfig = {
+            let config = include_str!("../../../ci_testing_config.ron");
```
Comment (mockersf) [inline crates/bevy_dev_tools/src/ci_testing/mod.rs:38]:
```suggestion
            let config = include_str!("../../../../ci_testing_config.ron");
```
... probably

Comment (BD103) [reply crates/bevy_dev_tools/src/ci_testing/mod.rs]: That looks right, since I moved it one directory deeper. Hopefully it passes this time :)

**Pull request merged**